llvm/lib/CodeGen/ScheduleDAGInstrs.cpp

```diff
@@ -69,6 +69,14 @@ static cl::opt<bool>
 static cl::opt<bool> UseTBAA("use-tbaa-in-sched-mi", cl::Hidden,
     cl::init(true), cl::desc("Enable use of TBAA during MI DAG construction"));
 
+static cl::opt<bool>
+    EnableSchedModel("schedmodel", cl::Hidden, cl::init(true),
+                     cl::desc("Use TargetSchedModel for latency lookup"));
+
+static cl::opt<bool>
+    EnableSchedItins("scheditins", cl::Hidden, cl::init(true),
+                     cl::desc("Use InstrItineraryData for latency lookup"));
```

Comment:
These are mutually exclusive though? What happens if you set both?

Reply:
As far as I can tell -- I don't see this mutual exclusion constraint encoded. It seems like the API handles the case of having neither --

Default if both are missing --
https://github.com/llvm/llvm-project/blob/173ec728d287f989053bd77521b3497b069c98a2/llvm/lib/CodeGen/TargetSchedule.cpp#L179

Default if both are missing --
https://github.com/llvm/llvm-project/blob/173ec728d287f989053bd77521b3497b069c98a2/llvm/lib/CodeGen/TargetSchedule.cpp#L269